app/models/service/dialog_properties.rb

```diff
@@ -0,0 +1,21 @@
+class Service
+  class DialogProperties
+    require_nested :Retirement
+
+    def initialize(options, user)
+      @options = options || {}
+      @user = user
+    end
+
+    def self.parse(options, user)
+      new(options, user).parse
+    end
+
+    def parse
+      Service::DialogProperties::Retirement.parse(@options, @user).tap do |attributes|
+        @attributes[:name] = @options['dialog_service_name'] if @options['dialog_service_name'].present?
```

Comment:
@gmcculloug 
The local attributes is not being used, should @attributes be attributes here.